pyomo/contrib/simplification/build.py

```diff
@@ -0,0 +1,95 @@
+from pybind11.setup_helpers import Pybind11Extension, build_ext
+from pyomo.common.fileutils import this_file_dir, find_library
+import os
+from distutils.dist import Distribution
+import sys
+import shutil
+import glob
+import tempfile
+from pyomo.common.envvar import PYOMO_CONFIG_DIR
+
+
+def build_ginac_interface(args=[]):
+    dname = this_file_dir()
+    _sources = ['ginac_interface.cpp']
+    sources = list()
+    for fname in _sources:
+        sources.append(os.path.join(dname, fname))
+
+    ginac_lib = find_library('ginac')
+    print(ginac_lib)
+    if ginac_lib is None:
+        raise RuntimeError(
+            'could not find GiNaC library; please make sure it is in the LD_LIBRARY_PATH environment variable'
+        )
+    ginac_lib_dir = os.path.dirname(ginac_lib)
+    ginac_build_dir = os.path.dirname(ginac_lib_dir)
+    ginac_include_dir = os.path.join(ginac_build_dir, 'include')
+    if not os.path.exists(os.path.join(ginac_include_dir, 'ginac', 'ginac.h')):
+        raise RuntimeError('could not find GiNaC include directory')
+
+    cln_lib = find_library('cln')
+    if cln_lib is None:
+        raise RuntimeError(
+            'could not find CLN library; please make sure it is in the LD_LIBRARY_PATH environment variable'
+        )
+    cln_lib_dir = os.path.dirname(cln_lib)
+    cln_build_dir = os.path.dirname(cln_lib_dir)
+    cln_include_dir = os.path.join(cln_build_dir, 'include')
+    if not os.path.exists(os.path.join(cln_include_dir, 'cln', 'cln.h')):
+        raise RuntimeError('could not find CLN include directory')
+
+    extra_args = ['-std=c++11']
+    ext = Pybind11Extension(
+        'ginac_interface',
+        sources=sources,
+        language='c++',
+        include_dirs=[cln_include_dir, ginac_include_dir],
+        library_dirs=[cln_lib_dir, ginac_lib_dir],
+        libraries=['cln', 'ginac'],
+        extra_compile_args=extra_args,
+    )
+
+    class ginac_build_ext(build_ext):
```

Comment:
Nitpicking - can you make this camelCase (ginarBuildExt) since it's a class?

Reply:
Done.